scale-tests/run_spark_scale_test.sh

```diff
@@ -0,0 +1,226 @@
+  --consumer-must-fail \
+  --consumer-write-to-cassandra \
+  --consumer-batch-size-seconds $CONSUMER_BATCH_SIZE_SECONDS \
+  --consumer-spark-cores-max $CONSUMER_SPARK_CORES_MAX \
+  --consumer-spark-executor-cores $CONSUMER_SPARK_EXECUTOR_CORES
+
+### Deploy finite jobs
+./scale-tests/kafka_cassandra_streaming_test.py \
+  $DISPATCHERS_FINITE_JSON_FILE \
+  $INFRASTRUCTURE_JSON_FILE \
+  $SUBMISSIONS_FINITE_JSON_FILE \
+  --jar $KAFKA_JAR \
+  --num-producers-per-kafka $NUM_PRODUCERS_PER_FINITE_KAFKA \
+  --num-consumers-per-producer $NUM_CONSUMERS_PER_FINITE_PRODUCER \
+  --producer-number-of-words $NUM_PRODUCER_WORDS \
+  --producer-words-per-second $NUM_PRODUCER_WORDS_PER_SECOND \
+  --producer-spark-cores-max $PRODUCER_SPARK_CORES_MAX \
+  --producer-spark-executor-cores $PRODUCER_SPARK_EXECUTOR_CORES \
+  --consumer-write-to-cassandra \
+  --consumer-batch-size-seconds $CONSUMER_BATCH_SIZE_SECONDS \
+  --consumer-spark-cores-max $CONSUMER_SPARK_CORES_MAX \
+  --consumer-spark-executor-cores $CONSUMER_SPARK_EXECUTOR_CORES
+    
+### Deploy infinite jobs
+INFINITE_PRODUCER_NUMBER_OF_WORDS=0
+
+./scale-tests/kafka_cassandra_streaming_test.py \
+  $DISPATCHERS_INFINITE_JSON_FILE \
+  $INFRASTRUCTURE_JSON_FILE \
+  $SUBMISSIONS_FINITE_JSON_FILE \
+  --jar $KAFKA_JAR \
+  --num-producers-per-kafka $NUM_PRODUCERS_PER_INFINITE_KAFKA \
+  --num-consumers-per-producer $NUM_CONSUMERS_PER_INFINITE_PRODUCER \
+  --producer-number-of-words $INFINITE_PRODUCER_NUMBER_OF_WORDS \
+  --producer-words-per-second $NUM_PRODUCER_WORDS_PER_SECOND \
+  --producer-spark-cores-max $PRODUCER_SPARK_CORES_MAX \
+  --producer-spark-executor-cores $PRODUCER_SPARK_EXECUTOR_CORES \
+  --consumer-batch-size-seconds $CONSUMER_BATCH_SIZE_SECONDS \
+  --consumer-spark-cores-max $CONSUMER_SPARK_CORES_MAX \
+  --consumer-spark-executor-cores $CONSUMER_SPARK_EXECUTOR_CORES
+
+## Deploy GPU jobs
+git checkout st-gpu-strict
+nohup ./scale-tests/batch_test.py $DISPATCHERS_JSON_FILE \
+    --docker-image "samvantran/spark-dcos-gpu:minpatch-nv390-cuda9-cudnn7115-tf15" \
+    --max-num-dispatchers 2 \
+    --submits-per-min 10 \
+    --spark-cores-max 4 \
+    --spark-mesos-executor-gpus 4 \
+    --spark-mesos-max-gpus 4 \
+    --no-supervise &
+
+git checkout master
+
+## Deploy batch jobs via marathon app
+./scale-tests/batch_test.py ${DISPATCHERS_JSON_FILE} --submits-per-min 4
+
+# The following is commented out due to a bug in the marathon app
+# Also, I'm not sure how to upload files to s3 in here 
+#
+#cd scale-tests/
+# ./deploy-marathon-service.sh spark-batch-workload.json.template \
+#   $CLUSTER_USERNAME \
+#   $CLUSTER_PASSWORD \
+#   "https://${TEST_S3_BUCKET}.s3.amazonaws.com/${TEST_S3_FOLDER}/${DISPATCHER_JSON_FILE}" \
+#   6 13000 "${DISPATCHER_JSON_FILE} --submits-per-min 4"
+# cd ..
+
+###### EOF
+
+## Run script inside stable container
+docker exec --net=host \
+  -v $(pwd):/spark-build \
+  -v $SSH_KEY:/ssh/key:ro \
+  -v ~/.aws./credentials:~/.aws/credentials \
```

Comment:
When mounting file paths to docker, absolute paths must be used. See the way the credential file is mounted into `test.sh`, for example. It may also be a good idea to mount this `:ro` (read-only).